msg.go

```diff
@@ -403,35 +403,31 @@ Loop:
 	return string(s), off1, nil
 }
 
-func packTxt(txt []string, msg []byte, offset int, tmp []byte) (int, error) {
+func packTxt(txt []string, msg []byte, offset int) (int, error) {
 	var err error
+	if offset >= len(msg) {
+		return offset, ErrBuf
+	}
 	if len(txt) == 0 {
-		if offset >= len(msg) {
-			return offset, ErrBuf
-		}
 		msg[offset] = 0
 		return offset, nil
```

Comment:
[ Quoting <notifications@github.com> in "Re: [miekg/dns] Remove temporary bu..." ]
>>  	if len(txt) == 0 {
>> -		if offset >= len(msg) {
>> -			return offset, ErrBuf
>> -		}
>>  		msg[offset] = 0
>>  		return offset, nil
>
>Is this the correct behavior when len(txt) == 0?
>
>Is this any different from
>```
>if len(txt) == 0 {
>    return offset, nil
>}
>```
>without doing the 0 assignment to the returned offset?

yes, you need to see the length of the txt in pkt to 0 to signal; no text